unable to build pam_pkcs11

```
tar -xvf pam_pkcs11-0.6.9.tar.gz 
cd pam_pkcs11-0.6.9/
./configure 
make
make  all-recursive
make[1]: Entering directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9'
Making all in .
make[2]: Entering directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9'
make[2]: Leaving directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9'
Making all in po
make[2]: Entering directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9/po'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9/po'
Making all in doc
make[2]: Entering directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9/doc'
xsltproc \
--stringparam  section.autolabel 1 \
--stringparam  section.label.includes.component.label 1 \
-o mappers_api.html pam_pkcs11.xsl mappers_api.xml
warning: failed to load external entity "http://docbook.sourceforge.net/release/xsl/current/xhtml/docbook.xsl"
compilation error: file pam_pkcs11.xsl line 3 element import
xsl:import : unable to load http://docbook.sourceforge.net/release/xsl/current/xhtml/docbook.xsl
make[2]: *** [mappers_api.html] Error 5
make[2]: Leaving directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9/doc'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/amitkuma2/Desktop/tak_print/pam_pkcs11-0.6.9'
make: *** [all] Error 2`
```

$ make 
make[3]: Entering directory '/root/pam_pkcs11/src/scconf'
  CC       scconf.lo
  CC       parse.lo
  CC       write.lo
  CC       sclex.lo
  CCLD     libscconf.la
make[3]: Leaving directory '/root/pam_pkcs11/src/scconf'
Making all in common
make[3]: Entering directory '/root/pam_pkcs11/src/common'
Making all in .
make[4]: Entering directory '/root/pam_pkcs11/src/common'
  CC       libcommon_la-algorithm.lo
  CC       libcommon_la-cert_vfy.lo
cert_vfy.c: In function 'verify_crl':
cert_vfy.c:146:15: error: storage size of 'obj' isn't known
   X509_OBJECT obj;
               ^~~
cert_vfy.c:159:3: warning: implicit declaration of function 'X509_OBJECT_free_contents'; did you mean 'X509_OBJECT_up_ref_count'? [-Wimplicit-function-declaration]
   X509_OBJECT_free_contents(&obj);
   ^~~~~~~~~~~~~~~~~~~~~~~~~
   X509_OBJECT_up_ref_count

--- a/src/common/cert_vfy.c
+++ b/src/common/cert_vfy.c
@@ -143,12 +143,17 @@ static X509_CRL *download_crl(const char *uri)
 static int verify_crl(X509_CRL * crl, X509_STORE_CTX * ctx)
 {
   int rv;
-  X509_OBJECT obj;
   EVP_PKEY *pkey = NULL;
   X509 *issuer_cert;
 
   /* get issuer certificate */
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
+  X509_OBJECT obj;
   rv = X509_STORE_get_by_subject(ctx, X509_LU_X509, X509_CRL_get_issuer(crl), &obj);
+#else
+  X509_OBJECT *obj = X509_OBJECT_new();
+  rv = X509_STORE_get_by_subject(ctx, X509_LU_X509, X509_CRL_get_issuer(crl), obj);
+#endif
   if (rv <= 0) {
     set_error("getting the certificate of the crl-issuer failed");
     return -1;
@@ -156,7 +161,11 @@ static int verify_crl(X509_CRL * crl, X509_STORE_CTX * ctx)
   /* extract public key and verify signature */
   issuer_cert = X509_OBJECT_get0_X509((&obj));
   pkey = X509_get_pubkey(issuer_cert);
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
   X509_OBJECT_free_contents(&obj);
+#else
+  X509_OBJECT_free(obj);
+#endif
   if (pkey == NULL) {
     set_error("getting the issuer's public key failed");
     return -1;
@@ -202,14 +211,17 @@ static int verify_crl(X509_CRL * crl, X509_STORE_CTX * ctx)
 static int check_for_revocation(X509 * x509, X509_STORE_CTX * ctx, crl_policy_t policy)
 {
   int rv, i, j;
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
   X509_OBJECT obj;
+#else
+  X509_OBJECT *obj = X509_OBJECT_new();
+#endif
   X509_REVOKED *rev = NULL;
   STACK_OF(DIST_POINT) * dist_points;
   DIST_POINT *point;
   GENERAL_NAME *name;
   X509_CRL *crl;
   X509 *x509_ca = NULL;
-  EVP_PKEY crl_pkey;
 
   DBG1("crl policy: %d", policy);
   if (policy == CRLP_NONE) {
@@ -227,13 +239,21 @@ static int check_for_revocation(X509 * x509, X509_STORE_CTX * ctx, crl_policy_t 
   } else if (policy == CRLP_OFFLINE) {
     /* OFFLINE */
     DBG("looking for an dedicated local crl");
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
     rv = X509_STORE_get_by_subject(ctx, X509_LU_CRL, X509_get_issuer_name(x509), &obj);
+#else
+    rv = X509_STORE_get_by_subject(ctx, X509_LU_CRL, X509_get_issuer_name(x509), obj);
+#endif
     if (rv <= 0) {
       set_error("no dedicated crl available");
       return -1;
     }
     crl = X509_OBJECT_get0_X509_CRL((&obj));
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
     X509_OBJECT_free_contents(&obj);
+#else
+    X509_OBJECT_free(obj);
+#endif
   } else if (policy == CRLP_ONLINE) {
     /* ONLINE */
     DBG("extracting crl distribution points");
@@ -247,7 +267,11 @@ static int check_for_revocation(X509 * x509, X509_STORE_CTX * ctx, crl_policy_t 
       }
       x509_ca = X509_OBJECT_get0_X509((&obj));
       dist_points = X509_get_ext_d2i(x509_ca, NID_crl_distribution_points, NULL, NULL);
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
       X509_OBJECT_free_contents(&obj);
+#else
+      X509_OBJECT_free(obj);
+#endif
       if (dist_points == NULL) {
         set_error("neither the user nor the ca certificate does contain a crl distribution point");
         return -1;

